GLMakie/src/GLAbstraction/GLTexture.jl

```diff
@@ -137,6 +137,9 @@ Colors from Colors.jl should mostly work as well
 Texture(image::Array{T, NDim}; kw_args...) where {T <: GLArrayEltypes, NDim} =
     Texture(pointer(image), size(image); kw_args...)::Texture{T, NDim}
 
+Texture(image::Adjoint{T, Array{T, NDim}}; kw_args...) where {T <: GLArrayEltypes, NDim} =
```

Comment:
should this be `image::AbstractArray{T, NDim}` for the most generality?

Reply:
Thank you for your review.

Originally this PR updated the `image::Array{T, NDim}` constructor to exactly what you write. But @SimonDanisch shed light on why this is wrong in his code review above. Would keeping the old `image::Array{T, NDim}` and adding an extra `image::AbstractArray{T, NDim}` constructor where I'd `collect` whatever the concrete `AbstractArray` subtype is make things safe? (Actually, probably not `collect` in the more general case, but the proposal from the first review: `Texture(image::AbstractArray{T, NDim}) = Texture(convert(Array{T, NDIM}, image))` - but the question remains, is this safe to do for any `AbstractArray`?)

You are of course right that more could be fixed in this PR now that I think about it more, this is another example that throws error of the same kind:

```julia
using GLMakie
x = rand(Float32, 5, 5);
y = @view x[2:3, 2:3];
heatmap(y)
```
```
Error showing value of type Makie.FigureAxisPlot:
ERROR: MethodError: no method matching GLMakie.GLAbstraction.Texture(::SubArray{Float32, 2, Matrix{Float32}, Tuple{UnitRange{Int64}, UnitRange{Int64}}, false}; minfilter=:nearest)
```
And the result of joining these two types really is an `AbstractArray`:

```
julia> typejoin(SubArray, Adjoint)
AbstractArray
```